src/rust/engine/src/nodes.rs

```diff
@@ -902,34 +902,49 @@ impl Task {
           .core
           .rule_graph
           .edges_for_inner(&entry)
-          .ok_or_else(|| throw(&format!("no edges for task {:?} exist!", entry)))
+          .ok_or_else(|| throw(&format!("No edges for task {:?} exist!", entry)))
           .and_then(|edges| {
-            edges
-              .entry_for(&dependency_key)
-              .cloned()
-              .ok_or_else(|| match get.input_type {
-                Some(ty) if externs::is_union(ty) => {
-                  let value = externs::type_for_type_id(ty).into_object();
-                  match externs::call_method(
-                    &value,
-                    "non_member_error_message",
-                    &[externs::val_for(&get.input)],
-                  ) {
-                    Ok(err_msg) => throw(&externs::val_to_str(&err_msg)),
-                    // If the non_member_error_message() call failed for any reason,
-                    // fall back to a generic message.
-                    Err(_e) => throw(&format!(
-                      "Type {} is not a member of the {} @union",
-                      get.input.type_id(),
-                      ty
-                    )),
-                  }
+            edges.entry_for(&dependency_key).cloned().ok_or_else(|| {
```

Comment:
This has gotten pretty large, and would be good to extract into a helper function.